test.ts

```diff
@@ -22,6 +22,25 @@ test('should parse JSON body', async () => {
   }).expect(200, { hello: 'world' })
 })
 
+test('should ignore JSON empty body', async () => {
+  const server = createServer(async (req: ReqWithBody, res) => {
+    await json()(req, res, (err) => void err && console.log(err))
+
+    res.setHeader('Content-Type', 'application/json')
+
+    res.end(JSON.stringify({ok: true}));
+  })
+
+  await makeFetch(server)('/', {
+    body: '',
```

Comment:
Maybe you could also test the case where no body is set at all. In think your code handles it as well.

Reply:
Hey, took me long to recheck this PR but I included the other test case too